src/middlewared/middlewared/plugins/disk_/configure_swap.py

```diff
@@ -0,0 +1,223 @@
+                                remove = True
+                            elif f'{part}.eli' in existing_swap_devices['partitions']:
+                                part = f'{part}.eli'
+                                remove = True
+
+                        if remove:
+                            await self.middleware.call('disk.swaps_remove_disks', [all_partitions[p]['disk']])
+                            existing_swap_devices['partitions'].remove(part)
+                except Exception:
+                    self.logger.warn('Failed to remove disk from swap', exc_info=True)
+                    # If something failed here there is no point in trying to create the mirror
+                    continue
+
+                part_a, part_b = part_ab
+
+                if not IS_LINUX and not dumpdev:
+                    dumpdev = await self.middleware.call('disk.dumpdev_configure', part_a)
+                name = await self.middleware.call('disk.new_swap_name')
+                if not name:
+                    # Which means maximum has been reached and we can stop
+                    break
+                part_a_path, part_b_path = all_partitions[part_a]['path'], all_partitions[part_b]['path']
+                try:
+                    await self.middleware.call(
+                        'disk.create_mirror', name, {
+                            'paths': [part_a_path, part_b_path],
+                            'extra': {'level': 1} if IS_LINUX else {},
+                        }
+                    )
+                except CallError:
+                    self.logger.warning('Failed to create swap mirror %s: %s', name)
+                    continue
+
+                swap_device = os.path.realpath(os.path.join('/dev/md', name)) if IS_LINUX else f'mirror/{name}'
+                create_swap_devices[swap_device] = {
+                    'path': swap_device,
+                    'encrypted_provider': None,
+                }
+
+            # Add remaining partitions to unused list
+            unused_partitions += partitions
+
+        if unused_partitions and not create_swap_devices and all(
+            not existing_swap_devices[k] for k in existing_swap_devices
+        ):
+            if not IS_LINUX and not dumpdev:
+                await self.middleware.call('disk.dumpdev_configure', unused_partitions[0])
+            swap_device = all_partitions[unused_partitions[0]]['path'] if IS_LINUX else unused_partitions[0]
+            create_swap_devices[swap_device] = {
+                'path': swap_device,
+                'encrypted_provider': all_partitions[unused_partitions[0]]['encrypted_provider'],
+            }
+
+        created_swap_devices = []
+        for name, data in create_swap_devices.items():
+            if IS_LINUX:
+                if not data['encrypted_provider']:
+                    cp = await Popen(
+                        f'echo "y" | cryptsetup -d /dev/urandom open --type plain {name} {name.split("/")[-1]}',
```

Comment:
`asyncio.create_subprocess_exec` has `stdin` argument, let's use it instead of the shell.